include/mull/Reporters/SourceManager.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "mull/SourceLocation.h"
+
+#include <map>
+#include <utility>
+#include <vector>
+
+namespace mull {
+
+struct LineOffset {
+  FILE *file;
+  std::vector<uint32_t> offsets;
+  LineOffset(FILE *file, std::vector<uint32_t> offsets)
+      : file(file), offsets(std::move(offsets)) {}
+};
+
+class SourceManager {
+public:
+  std::string getLine(const SourceLocation &location);
+
+  ~SourceManager();
+
+private:
+  std::map<std::string, LineOffset> lineOffsets;
+
+  LineOffset &getLineOffset(const SourceLocation &location);
+};
+
+} // namespace mull
```

Comment:
Would you mind submitting this as a separate pull request? It makes sense to extract it to an interface.